homeassistant/components/mqtt/image.py

```diff
@@ -0,0 +1,199 @@
+"""Image that loads a picture from using an MQTT topic."""
+from __future__ import annotations
+
+from base64 import b64decode
+from collections.abc import Callable
+import functools
+import logging
+import ssl
+
+import httpx
+import voluptuous as vol
+
+from homeassistant.components import image
+from homeassistant.components.image import DEFAULT_CONTENT_TYPE, ImageEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_NAME, CONF_VALUE_TEMPLATE
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.service_info.mqtt import ReceivePayloadType
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.util import dt as dt_util
+
+from . import subscription
+from .config import MQTT_BASE_SCHEMA
+from .const import CONF_ENCODING, CONF_QOS, CONF_TOPIC
+from .debug_info import log_messages
+from .mixins import MQTT_ENTITY_COMMON_SCHEMA, MqttEntity, async_setup_entry_helper
+from .models import MqttValueTemplate, ReceiveMessage
+from .util import get_mqtt_data, valid_subscribe_topic
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_CONTENT_TYPE = "content_type"
+CONF_IMAGE_ENCODING = "image_encoding"
+CONF_FROM_URL = "from_url"
+
+DEFAULT_NAME = "MQTT Image"
+
+MQTT_IMAGE_ATTRIBUTES_BLOCKED = frozenset({})  # type: ignore[var-annotated]
+
+
+def validate_content_type(content_type: str) -> str:
+    """Validate the config type is an image."""
+    if content_type.split("/")[0] != "image":
+        raise vol.Invalid(f"Content type {content_type} is not valid")
+    return content_type
+
+
+PLATFORM_SCHEMA_BASE = MQTT_BASE_SCHEMA.extend(
+    {
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Required(CONF_TOPIC): valid_subscribe_topic,
+        vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+        vol.Optional(CONF_IMAGE_ENCODING): "b64",
+        vol.Optional(CONF_CONTENT_TYPE, default=DEFAULT_CONTENT_TYPE): vol.All(
+            cv.string, validate_content_type
+        ),
+        vol.Optional(CONF_FROM_URL, default=False): cv.boolean,
```

Comment:
Instead of this, how about allowing configuring either a URL topic or a image data topic?

Reply:
Hmm, may be that is more straight through indeed. Any how we need an extra config topic.